### PR TITLE
Licence Ja Rule under the LGPL

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: tests/doxygen/overview.dox [hunk 1]
```diff
@@ -49,5 +49,9 @@
  *
  * @section sec_Licensing Licensing
  *
- * TODO(simon): complete this
+ * The Ja Rule codebase is licensed under the
```
Comment (peternewman) [inline tests/doxygen/overview.dox]: Isn't it Ja-Rule?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,13 @@
 # Ja Rule
```
Comment (peternewman) [inline README.md]: Ja-Rule?

File: README.md [hunk 3]
```diff
@@ -23,7 +18,10 @@ buffer would suffice.
 For DMX / RDM messages, the response message is sent when the transceiver
 completes the transaction, so the host received positive acknowledgement.
 
+## Licensing
 
-## TODO:
+The Ja Rule codebase is licensed under the
```
Comment (peternewman) [inline README.md]: Ja-Rule?

**Comment by peternewman** [comment]
LGTM. We should probably add the licenses in the top level directory too shouldn't we?

**Comment by peternewman** [comment]
It would be good to port the enforcer tweaks back to ola too.

**Comment by peternewman** [comment]
We need a LICENSE file in firmware/src for the enforcer.

File: README.md (hunk 2 above)
Comment (nomis52) [inline README.md]: I believe not: https://en.wikipedia.org/wiki/Ja_Rule :)

Comment (peternewman) [inline README.md]: Hmm, I thought it was just nonsense words from GitHub.

File: tests/scripts/enforce_licence.py [hunk 4]
```diff
@@ -159,8 +166,16 @@ def GetDirectoryLicences(root_dir):
...
     licence = licences.get(dir_name)
-    if licence is not None:
-      for sub_dir in subdirs:
+    if licence is None:
+      continue
+
+    for sub_dir in subdirs:
+      full_path = os.path.join(dir_name, sub_dir)
+      relative_path = os.path.relpath(full_path, root_dir)
+      ok = True
```
Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Can you give it a more obvious variable name please.

**Comment by peternewman** [comment]
Just one outstanding issue.

**Pull request merged**